### PR TITLE
[js/rn] Fix React Native CI pipeline E2E test

### PR DESCRIPTION
**Pull request by YUNQIUGUO**
### Description

Based on this kindly provided quick fix: https://github.com/microsoft/onnxruntime/pull/16411

See more description in the above linked pr about bumping AGP version, etc.

Also fixed import header file path in detox e2e test.

### Motivation and Context

Good build:
https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=1041757&view=logs&j=de302ec2-2305-57e0-e8c6-cd89c569f2a3&t=9894c870-b8ce-548d-51ff-8f44d21a4117&l=18

### CODE REVIEW

File: js/react_native/ios/OnnxruntimeModule.mm [hunk 1]
```diff
@@ -22,7 +22,7 @@
 #include "onnxruntime_extensions.h"
 #else
 #include "onnxruntime/coreml_provider_factory.h"
-#include "onnxruntime/onnxruntime_cxx_api.h"
+#import <onnxruntime/onnxruntime_cxx_api.h>
```
Comment (edgchen1) [inline js/react_native/ios/OnnxruntimeModule.mm:25]: why do we use `#import` here? it's not an Objective-C header.

Comment (YUNQIUGUO) [reply js/react_native/ios/OnnxruntimeModule.mm:25]: yea, so I am not 100% sure yet how the ios project header file path all wired up, but from the code history, we used to have this https://github.com/microsoft/onnxruntime/pull/15800/files#diff-6cd96aa3c0515756787745b690d4b9fae90a00e27bbdcc289f4b825f11979f3dL9 syntax for including headers in the js/react_native/ios which works with xcodebuild.

Comment (YUNQIUGUO) [reply js/react_native/ios/OnnxruntimeModule.mm:25]: tried on another branch `#include` with angle brackets would work too. will update.

**Comment by YUNQIUGUO** [comment]
looks like the `onnxruntime_cxx_api.h` header search path works with a released pod, but still failed at a local-built version `onnxruntime-c` pod.

To unblock the pipeline failure, I temporarily commented out the lines to use local built onnxruntime-c pod in react-native-ci.yml file and I think we can check in the other fixes first.

**Review by jchen351 — APPROVED** [review]

**Pull request merged**